blog/2015/09/teaching-to-the-workflow.html

```diff
@@ -0,0 +1,91 @@
+      this together. This is why I think it's so important for
+      instructors to screw up when teaching and <em>slowly</em> talk
+      through how they untangle themselves. No grad student or
+      programmer starts out like Athena from the head of Zeus, fully
+      formed and brilliant from the beginning. Because writing,
+      documenting, and openly sharing scientific thinking and the code
+      behind it&mdash;even with mistakes&mdash;is one of the best ways
+      we have as scientists to be honest with ourselves and with
+      others.
+    </p>
+  </li>
+  <li>
+    <p>
+      The idea of provenance. Not just in the formal sense but also in
+      the broader sense, related to a whole ton of important ideas,
+      including but not limited to: metadata, version control, code
+      modularity, Don't Repeat Yourself, testing, open access,
+      etc. Science is deeply and fundamentally about provenance of
+      ideas and provenance of evidence. All scientists get this
+      intuitively, but the ways to document this trail are changing
+      and multiplying rapidly. Where is the idea from (articles, DOIs,
+      open access)? What is the evidence for it (figures, tables,
+      statistics)? How were the analyses done (code, version control,
+      formal computational provenance)? Where did the data come from
+      (metadata, data archiving, well-documented field or lab
+      methods)? Sofwtare Carpentry has for a long time taught people
+      how to structure and document code and make analyses
+      reproducible, but I think a lot of people (myself included) also
+      want and need to learn how to better structure, from start to
+      finish, their project organizational schemes and filesystems
+      (and their science!) with provenance in mind. Because the
+      process of science is messy, it's not and will never be
+      easy. But it seems to me that many of us want to know how to do
+      it better.
+    </p>
+  </li>
+  <li>
+    <p>
+      The idea of structured data. This could be anything from how to
+      use Excel responsibly, to munging/slicing/dicing in python/R, to
+      designing bespoke SQLite databases, but the <em>idea</em> of it
+      is crucial. This is very rarely taught in any formal setting and
+      absolutely is the key enabling anything analytical that
+      follows. Most people, in my experience, struggle through until
+      they, by chance, hit upon the format that enables pivot
+      tables/JMP/ggplot to work well. Structuring data well from the
+      start leads to easier and less frustrating analyses, easier and
+      less frustrating collaborations, and easier and less frustrating
+      archiving. In the end, science and scientists win when data are
+      well-structured. Format code for people and format data for
+      machines.
+    </p>
+  </li>
+</ul>
+<p>
+  What would <em>you</em> teach to prepare people to do science the
```

Comment:
Change to: What would <em>you</em> teach to prepare people to do the best science possible?